rules/Php80/Rector/ClassMethod/AddParamBasedOnParentClassMethodRector.php

```diff
@@ -259,6 +260,12 @@ private function resolveParamType(Param $param): null|Identifier|Name|ComplexTyp
         $paramType = $param->type;
         $paramType->setAttribute(AttributeKey::ORIGINAL_NODE, null);
 
+        // this is needed to avoid reprint of invalid tokens
+        // @see https://github.com/rectorphp/rector/issues/8712
+        if ($paramType instanceof NullableType) {
+            $paramType->type->setAttribute(AttributeKey::ORIGINAL_NODE, null);
+        }
```

Comment:
This is the trick... sometimes, the whole node tree from top to bottom original node has to be set `null`.

Maybe we'll need something similar for union/complex types later.

Reply:
There is `$this->nodeFactory->createReprintedExpr()` for that

Reply:
It seems `createReprintedExpr()` should allow any Node instead of just Expr

https://github.com/rectorphp/rector-src/blob/2a14c30a574d0c539ba289fd1447accd4a15ee17/src/PhpParser/Node/NodeFactory.php#L340

Reply:
Good point :+1: